app/modules/collaborations/models.py

```diff
@@ -17,13 +51,161 @@ class Collaboration(db.Model, HoustonModel):
     guid = db.Column(
         db.GUID, default=uuid.uuid4, primary_key=True
     )  # pylint: disable=invalid-name
-    title = db.Column(db.String(length=50), nullable=False)
+    title = db.Column(db.String(length=50), nullable=True)
+
+    collaboration_user_associations = db.relationship(
+        'CollaborationUserAssociations', back_populates='collaboration'
+    )
+
+    def __init__(
+        self, user_guids=None, approval_states=None, is_initiator=None, **kwargs
+    ):
+        # I think this can be adapted to allow >2 User collabs, opening the door for Org-level.
+        if user_guids is None or len(user_guids) != 2:
+            raise ValueError(
+                '__init__ Collaboration: Creating a collaboration requires a user_guids list.'
+            )
+
+        # You can create a collaboration where both users 'approved'
+        if approval_states is not None and len(approval_states) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of approval_states was not equal to provided user list.'
+            )
+
+        if is_initiator is not None and len(is_initiator) != len(user_guids):
+            raise ValueError(
+                '__init__ Collaboration: Length of is_initiator states was not equal to provided user list.'
+            )
+
+        for guid_index in range(len(user_guids)):
+            from app.modules.users.models import User
+
+            user_tup = User.ensure_edm_obj(user_guids[guid_index])
+
+            if user_tup is None or user_tup[0] is None:
+                raise ValueError(
+                    '__init__ Collaboration: One of the user_guids provided was had no associated user.'
+                )
+
+            collab_user_assoc = CollaborationUserAssociations(
+                collaboration=self, user=user_tup[0]
+            )
+
+            if (
+                approval_states is not None
+                and approval_states[guid_index] in CollaborationUserState.ALLOWED_STATES
+            ):
+                collab_user_assoc.read_approval_state = approval_states[guid_index]
+            if (
+                is_initiator is not None
+                and isinstance(is_initiator[guid_index], bool)
+                and is_initiator[guid_index] is True
+            ):
+                collab_user_assoc.initiator = is_initiator[guid_index]
+                # If you initiate you approve
+                log.warning(' setting APPROVED!')
```

Comment:
this feels like a log.info. It's certainly not a warning, it's normal operation

Reply:
That feels like a test log that that I accidentally left in there actually. Removing.